src/connection.toit

```
@@ -143,68 +150,72 @@ class Connection:
     if current_reader_: throw "Previous response not completed"
     if not socket_: return null
 
-    if not reader_.can_ensure 1:
+    reader := socket_.in
+
+    if not reader.try-ensure-buffered 1:
       if write_closed_: close
       return null
-    index_of_first_space := reader_.index_of_or_throw ' '
-    method := reader_.read_string (index_of_first_space)
-    reader_.skip 1
-    path := reader_.read_string (reader_.index_of_or_throw ' ')
-    reader_.skip 1
-    version := reader_.read_string (reader_.index_of_or_throw '\r')
-    reader_.skip 1
-    if reader_.read_byte != '\n': throw "FORMAT_ERROR"
+    index_of_first_space := reader.index_of ' ' --throw-if-missing
```

Looking at this, I think I would have preferred `--throw-if-absent`.